src/main/java/com/epam/page/object/generator/model/searchrule/SelenideSearchRule.java

```diff
@@ -0,0 +1,152 @@
+package com.epam.page.object.generator.model.searchrule;
+
+import com.codeborne.selenide.ElementsCollection;
+import com.epam.page.object.generator.error.XpathToCssTransformerException;
+import com.epam.page.object.generator.model.ClassAndAnnotationPair;
+import com.epam.page.object.generator.model.Selector;
+import com.epam.page.object.generator.model.webelement.CommonWebElement;
+import com.epam.page.object.generator.model.webelement.SelenideWebElement;
+import com.epam.page.object.generator.model.webelement.WebElement;
+import com.epam.page.object.generator.model.webgroup.CommonWebElementGroup;
+import com.epam.page.object.generator.model.webgroup.SelenideWebElementGroup;
+import com.epam.page.object.generator.model.webgroup.WebElementGroup;
+import com.epam.page.object.generator.util.SearchRuleType;
+import com.epam.page.object.generator.util.SelectorUtils;
+import com.epam.page.object.generator.util.XpathToCssTransformer;
+import com.epam.page.object.generator.validator.ValidationResult;
+import com.epam.page.object.generator.validator.ValidatorVisitor;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@link SelenideSearchRule} describes {@link SearchRule} with one of the type defined in property
+ * file in 'commonSearchRule' group.
+ */
+public class SelenideSearchRule implements SearchRule {
+
+    private String uniqueness;
+    private SearchRuleType type;
+    private Selector selector;
+    private ClassAndAnnotationPair classAndAnnotation;
+    private XpathToCssTransformer transformer;
+    private SelectorUtils selectorUtils;
+    private boolean annotation;
+
+    private List<ValidationResult> validationResults = new ArrayList<>();
+    private final static Logger logger = LoggerFactory.getLogger(SelenideSearchRule.class);
+
+    public SelenideSearchRule(String uniqueness, SearchRuleType type, Selector selector,
+                              ClassAndAnnotationPair classAndAnnotation,
+                              XpathToCssTransformer transformer,
+                              SelectorUtils selectorUtils,
+                              boolean annotation) {
+        this.uniqueness = uniqueness;
+        this.type = type;
+        this.selector = selector;
+        this.annotation = annotation;
+        this.classAndAnnotation = classAndAnnotation;
+        this.transformer = transformer;
+        this.selectorUtils = selectorUtils;
+    }
+
+    public String getUniqueness() {
+        return uniqueness;
+    }
+
+    public SearchRuleType getType() {
+        return type;
+    }
+
+    private String getRequiredValue(Element element) {
```

Comment:
Is it possible to avoid repeating this logic?

Reply:
I guess yes, but not now. It was disigned like that. If we have a task to refactor it (again), we can figure out what to do to avoid repeating the logic.

Reply:
When we were working on this task we find that we have to do a lot of copy-paste just to add a new Builder. Currently rewriting could take a lot of efforts.